src/Verify.EntityFramework.Tests/CoreTests.cs

```diff
@@ -327,6 +333,55 @@ await data.Companies
         #endregion
     }
 
+    [Test]
+    public async Task RecordingWebApplicationFactory()
+    {
+        var factory = new CustomWebApplicationFactory();
+
+        var httpClient = factory.CreateClient();
+
+        EfRecording.StartRecording();
+
+        _ = await httpClient.GetAsync("/companies");
+
+        var sql = EfRecording.FinishRecording();
+
+        await Verify(sql);
+    }
+
+    class CustomWebApplicationFactory : WebApplicationFactory<Startup>
+    {
+        protected override void ConfigureWebHost(IWebHostBuilder builder) =>
+            builder
+                .ConfigureTestServices(services =>
+                {
+                    var database = DbContextBuilder.GetDatabase("RecordingWebApplicationFactory").GetAwaiter().GetResult();
+                    var data = database.Context;
+
+                    services.AddScoped(_ => data);
+                });
+
+        protected override IHostBuilder CreateHostBuilder() =>
+            Host.CreateDefaultBuilder()
+                .ConfigureWebHostDefaults(webBuilder => webBuilder.UseStartup<Startup>());
+    }
+
+    public class Startup
```

Comment:
I tried to use minimal api but I didn't manage to solve it unless I created a new project, which seemed overkill. Unfortunately `WebApplicationFactory<T>` will be looking for the assembly entry point and there already is a (machine generated) entry point in test projects.